Integrations/Pwned-V2/Pwned-V2.py

```diff
@@ -33,22 +35,37 @@
 EMAIL_TRUNCATE_VERIFIED_SUFFIX = '?truncateResponse=false&includeUnverified=true'
 DOMAIN_TRUNCATE_VERIFIED_SUFFIX = '&truncateResponse=false&includeUnverified=true'
 
+RETRIES_END_TIME = datetime.min
+
 ''' HELPER FUNCTIONS '''
 
 
 def http_request(method, url_suffix, params=None, data=None):
-    res = requests.request(
-        method,
-        BASE_URL + url_suffix,
-        verify=USE_SSL,
-        params=params,
-        data=data,
-        headers=HEADERS
-    )
+    while True:
+        res = requests.request(
+            method,
+            BASE_URL + url_suffix,
+            verify=USE_SSL,
+            params=params,
+            data=data,
+            headers=HEADERS
+        )
+
+        if res.status_code != 429:
+            # Rate limit response code
+            break
+
+        if datetime.now() > RETRIES_END_TIME:
+            return_error('Max retry time has exceeded.')
+
+        wait_regex = re.search(r'\d+', res.json()['message'])
+        if wait_regex:
+            wait_amount = wait_regex.group()
+            time.sleep(int(wait_amount))
```

Comment:
Calculate if wait is going to exceed RETRIES_END_TIME. If so, don't wait and return an error immediately.